[Flaking tests] k8s.io/kubernetes/test/integration/auth - TestConstrainedImpersonation

### Which jobs are flaking?

- https://prow.k8s.io/job-history/gs/kubernetes-ci-logs/logs/ci-kubernetes-integration-master
- https://prow.k8s.io/job-history/gs/kubernetes-ci-logs/logs/ci-kubernetes-integration-arm64-master
- https://prow.k8s.io/job-history/gs/kubernetes-ci-logs/logs/ci-kubernetes-integration-ppc64le-master

### Which tests are flaking?

` k8s.io/kubernetes/test/integration/auth - TestConstrainedImpersonation`

### Since when has it been flaking?

Triage tool shows the earliest failures on[2044429976823926784 4/15/2026, 8:28:18 PM](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-integration-arm64-master/2044429976823926784) as per [Link](https://storage.googleapis.com/k8s-triage/index.html?text=TestConstrainedImpersonation)

### Testgrid link

https://testgrid.k8s.io/sig-release-master-blocking#integration-master
https://testgrid.k8s.io/sig-release-master-blocking#integration-arm64-master

### Reason for failure (if possible)

```
{Failed  === RUN   TestConstrainedImpersonation/impersonating_scheduled_node
    auth_test.go:1368: unexpected impersonation metrics diff (-want +got):   []string{
          	... // 9 identical elements
          	"apiserver_impersonation_authorization_attempts_duration_seconds_"...,
          	"apiserver_impersonation_authorization_attempts_duration_seconds_"...,
          	strings.Join({
          		"apiserver_impersonation_authorization_attempts_duration_seconds_",
          		`sum{decision="denied",mode="legacy"`,
        + 		"} 7e-06",
          		"} FP",
          	}, ""),
          	`apiserver_impersonation_authorization_attempts_total{decision="a`...,
          	`apiserver_impersonation_authorization_attempts_total{decision="d`...,
          	`apiserver_impersonation_authorization_attempts_total{decision="d`...,
          }
--- FAIL: TestConstrainedImpersonation/impersonating_scheduled_node (1.85s)

=== RUN   TestConstrainedImpersonation
--- FAIL: TestConstrainedImpersonation (10.58s)
}
```

### Anything else we need to know?

We were able to reproduce this flake locally using stress tool.
Steps to reproduce:

1. Install stress tool (# go install [golang.org/x/tools/cmd/stress@latest](http://golang.org/x/tools/cmd/stress@latest)).
2. git clone  https://github.com/kubernetes/kubernetes
3. cd kubernetes
4. hack/install-etcd.sh
5. export the output from above command (For example, export PATH="${PATH}:/root/kubernetes/third_party/etcd")
6. etcd &
7. go test ./test/integration/auth -c -count=1
8.  /root/go/bin/stress ./auth.test -test.run TestConstrainedImpersonation

```
[root@varsharani1 kubernetes]# uname -a
Linux varsharani1 5.14.0-691.el9.x86_64 #1 SMP PREEMPT_DYNAMIC Mon Apr 6 16:34:53 UTC 2026 x86_64 x86_64 x86_64 GNU/Linux
[root@varsharani1 kubernetes]# go test ./test/integration/auth -c -count=1
[root@varsharani1 kubernetes]# /root/go/bin/stress ./auth.test -test.run TestConstrainedImpersonation
5s: 0 runs so far, 0 failures, 8 active
10s: 0 runs so far, 0 failures, 8 active
15s: 0 runs so far, 0 failures, 8 active
20s: 0 runs so far, 0 failures, 8 active
25s: 8 runs so far, 0 failures, 8 active
30s: 8 runs so far, 0 failures, 8 active
35s: 8 runs so far, 0 failures, 8 active
40s: 8 runs so far, 0 failures, 8 active
45s: 12 runs so far, 0 failures, 8 active
50s: 16 runs so far, 0 failures, 8 active
55s: 16 runs so far, 0 failures, 8 active
1m0s: 16 runs so far, 0 failures, 8 active
1m5s: 18 runs so far, 0 failures, 8 active
1m10s: 24 runs so far, 0 failures, 8 active
1m15s: 24 runs so far, 0 failures, 8 active
1m20s: 24 runs so far, 0 failures, 8 active
1m25s: 25 runs so far, 0 failures, 8 active
1m30s: 31 runs so far, 0 failures, 8 active
1m35s: 32 runs so far, 0 failures, 8 active
1m40s: 32 runs so far, 0 failures, 8 active
1m45s: 32 runs so far, 0 failures, 8 active
1m50s: 37 runs so far, 0 failures, 8 active
1m55s: 40 runs so far, 0 failures, 8 active
2m0s: 40 runs so far, 0 failures, 8 active
2m5s: 40 runs so far, 0 failures, 8 active
2m10s: 43 runs so far, 0 failures, 8 active
2m15s: 48 runs so far, 0 failures, 8 active
2m20s: 48 runs so far, 0 failures, 8 active
2m25s: 48 runs so far, 0 failures, 8 active
2m30s: 51 runs so far, 0 failures, 8 active
2m35s: 56 runs so far, 0 failures, 8 active
2m40s: 56 runs so far, 0 failures, 8 active
2m45s: 56 runs so far, 0 failures, 8 active
2m50s: 57 runs so far, 0 failures, 8 active
2m55s: 63 runs so far, 0 failures, 8 active
3m0s: 64 runs so far, 0 failures, 8 active
3m5s: 64 runs so far, 0 failures, 8 active
3m10s: 65 runs so far, 0 failures, 8 active
3m15s: 69 runs so far, 0 failures, 8 active
3m20s: 72 runs so far, 0 failures, 8 active
3m25s: 72 runs so far, 0 failures, 8 active
3m30s: 72 runs so far, 0 failures, 8 active
3m35s: 75 runs so far, 0 failures, 8 active

/tmp/go-stress-20260427T042454-1634805402
I0427 04:28:10.391439 2343949 etcd.go:66] etcd already running at http://127.0.0.1:2379
W0427 04:28:10.671017 2343949 registry.go:342] setting componentGlobalsRegistry in SetFallback. We recommend calling componentGlobalsRegistry.Set() right after parsing flags to avoid using feature gates before their final values are set by the flags.
I0427 04:28:10.898899 2343949 serving.go:405] Generated self-signed cert (/tmp/test-integration-TestConstrainedImpersonation2405518444/apiserver.crt, /tmp/test-integration-TestConstrainedImpersonation2405518444/apiserver.key)
I0427 04:28:10.898937 2343949 options.go:263] external host was not specified, using 9.30.140.212
E0427 04:28:11.288776 2343949 options.go:205] The manifest file is empty, ignoring.
I0427 04:28:11.305088 2343949 shared_informer.go:402] "Waiting for caches to sync"
I0427 04:28:11.313551 2343949 shared_informer.go:402] "Waiting for caches to sync"
I0427 04:28:11.317060 2343949 plugins.go:157] Loaded 14 mutating admission controller(s) successfully in the following order: NamespaceLifecycle,LimitRanger,ServiceAccount,TaintNodesByCondition,Priority,DefaultTolerationSeconds,DefaultStorageClass,StorageObjectInUseProtection,PodGroupProtection,RuntimeClass,DefaultIngressClass,PodTopologyLabels,MutatingAdmissionPolicy,MutatingAdmissionWebhook.
I0427 04:28:11.317077 2343949 plugins.go:160] Loaded 17 validating admission controller(s) successfully in the following order: LimitRanger,ServiceAccount,PodSecurity,Priority,PersistentVolumeClaimResize,RuntimeClass,CertificateApproval,CertificateSigning,ClusterTrustBundleAttest,CertificateSubjectRestriction,PodGroupWorkloadExists,NodeDeclaredFeatureValidator,JobValidation,PodResizeValidator,ValidatingAdmissionPolicy,ValidatingAdmissionWebhook,ResourceQuota.
I0427 04:28:11.317325 2343949 instance.go:240] Using reconciler: lease
I0427 04:28:11.323596 2343949 cidrallocator.go:198] starting ServiceCIDR Allocator Controller
I0427 04:28:11.375456 2343949 handler.go:304] Adding GroupVersion  v1 to ResourceManager
I0427 04:28:11.37566
…
3m40s: 80 runs so far, 1 failures (1.25%), 8 active
3m45s: 80 runs so far, 1 failures (1.25%), 8 active
3m50s: 80 runs so far, 1 failures (1.25%), 8 active
3m55s: 82 runs so far, 1 failures (1.22%), 8 active
4m0s: 88 runs so far, 1 failures (1.14%), 8 active
```

### Relevant SIG(s)

/sig testing

--- a/test/integration/auth/auth_test.go
+++ b/test/integration/auth/auth_test.go
@@ -1452,7 +1452,7 @@ func assertImpersonationMetrics(t *testing.T, ctx context.Context, client client
 	}
 
 	var gotMetricStrings []string
-	trimFP := regexp.MustCompile(`(.*)(} \d+\.\d+.*)`)
+	trimFP := regexp.MustCompile(`(} )[\de.+-]+(.*)`)
 	for line := range strings.SplitSeq(string(body), "\n") {
 		if !strings.HasPrefix(line, "apiserver_impersonation_") {
 			continue
@@ -1462,7 +1462,7 @@ func assertImpersonationMetrics(t *testing.T, ctx context.Context, client client
 			continue
 		}
 		if strings.Contains(line, "_seconds_sum") {
-			line = trimFP.ReplaceAllString(line, `$1`) + "} FP"
+			line = trimFP.ReplaceAllString(line, "${1}FP")
 		}
 		gotMetricStrings = append(gotMetricStrings, line)
 	}

